classes/submission/Collector.php

```diff
@@ -623,8 +625,11 @@ public function getQueryBuilder(): Builder
         }
 
         if (isset($this->categoryIds)) {
-            $q->join('publication_categories as pc', 's.current_publication_id', '=', 'pc.publication_id')
-                ->whereIn('pc.category_id', $this->categoryIds);
+            $publicationIds = PublicationCategory::withCategoryIds($this->categoryIds)
+                        ->distinct()
+                        ->pluck('publication_id')->toArray();
+        
+            $q->whereIn('s.current_publication_id', $publicationIds);
```

Comment:
I'm open to ideas -- but I don't think it'll be possible (or even desirable) to completely isolate the storage implementation so that each model class doesn't include schema details on related entities. Inside the storage layer, it should be possible to quickly read/intuit the SQL query by reading the PHP code, and delegating parts of the implementation to the other entity classes makes it harder to read. And it'll be even harder to preserve that isolation when we need to optimize things. As you can see, delegating to the respective eloquent model is already causing data to be materialized into memory rather than letting the database handle things in a single operation.

It's *outside* the storage layer that I want the database details to be opaque. Here in the collectors and Eloquent objects, it's OK.

So my preference would be to continue with more or less the same syntax we had before -- it's OK if the submission collector names the `publication_categories` table and the columns in it, since we're already working inside the storage layer.

Reply:
Ok!